vizro-core/tests/e2e/vizro/dashboards/default/pages/actions_properties_shortcut_page.py

```diff
@@ -0,0 +1,166 @@
+    # Added to handle the bug when a page custom action is triggered if on-page-load is not defined
+    if not button_number_of_clicks:
+        raise PreventUpdate
+
+    # Sleep to see what part of the output component is updating
+    time.sleep(0.5)
+
+    title = f"Button clicked {button_number_of_clicks} times."
+
+    # -2 is set due to internal action outputs: "_action_finished" and "_action_progress_indicatior"
+    return [title] * (len(ctx.outputs_list) - 2)
+
+
+vm.Page.add_type("controls", vm.Button)
+
+figures_title_header_footer_page = vm.Page(
+    title=cnst.ACTION_PROPERTIES_SHORTCUT_TITLE_DESCRIPTION_HEADER_FOOTER_PAGE,
+    id=cnst.ACTION_PROPERTIES_SHORTCUT_TITLE_DESCRIPTION_HEADER_FOOTER_PAGE,
+    layout=vm.Grid(grid=[[0, 1]]),
+    components=[
+        vm.Graph(
+            id=cnst.ACTION_SHORTCUT_GRAPH_ID,
+            title=cnst.ACTION_PROPERTIES_SHORTCUT_DEFAULT_FIGURE_TEXT,
+            description=cnst.ACTION_PROPERTIES_SHORTCUT_DEFAULT_FIGURE_TEXT,
+            header=cnst.ACTION_PROPERTIES_SHORTCUT_DEFAULT_FIGURE_TEXT,
+            footer=cnst.ACTION_PROPERTIES_SHORTCUT_DEFAULT_FIGURE_TEXT,
+            figure=px.scatter(df_3_rows, x="sepal_width", y="sepal_length", color="species"),
+        ),
+        vm.AgGrid(
+            id=cnst.ACTION_SHORTCUT_AG_GRID_ID,
+            title=cnst.ACTION_PROPERTIES_SHORTCUT_DEFAULT_FIGURE_TEXT,
+            description=cnst.ACTION_PROPERTIES_SHORTCUT_DEFAULT_FIGURE_TEXT,
+            header=cnst.ACTION_PROPERTIES_SHORTCUT_DEFAULT_FIGURE_TEXT,
+            footer=cnst.ACTION_PROPERTIES_SHORTCUT_DEFAULT_FIGURE_TEXT,
+            figure=dash_ag_grid(df_3_rows),
+        ),
+    ],
+    controls=[
+        vm.Button(
+            id=cnst.ACTION_SHORTCUT_TRIGGER_BUTTON_ID,
+            actions=[
+                vm.Action(
+                    function=action_return_text(f"{cnst.ACTION_SHORTCUT_TRIGGER_BUTTON_ID}"),
+                    outputs=[
+                        # Graph
+                        f"{cnst.ACTION_SHORTCUT_GRAPH_ID}.title",
+                        f"{cnst.ACTION_SHORTCUT_GRAPH_ID}.description",
+                        f"{cnst.ACTION_SHORTCUT_GRAPH_ID}.header",
+                        f"{cnst.ACTION_SHORTCUT_GRAPH_ID}.footer",
+                        # AgGrid
+                        f"{cnst.ACTION_SHORTCUT_AG_GRID_ID}.title",
+                        f"{cnst.ACTION_SHORTCUT_AG_GRID_ID}.description",
+                        f"{cnst.ACTION_SHORTCUT_AG_GRID_ID}.header",
+                        f"{cnst.ACTION_SHORTCUT_AG_GRID_ID}.footer",
+                    ],
+                )
+            ],
+        ),
+    ],
+)
+
+
+@capture("action")
+def action_select_ag_grid_rows(button_number_of_clicks):
+    # Sleep to see what part of the output component is updating
+    time.sleep(0.5)
+
+    num_rows = button_number_of_clicks % 4
+    selected_rows = df_3_rows.iloc[:num_rows].to_dict("records") if num_rows > 0 else []
+
+    return selected_rows
```

Comment:
If you use the magic name `_trigger` you can skip defining the input. In that case the action's function definition would change from:

`function=action_select_ag_grid_rows(f"{cnst.BUTTON_AG_GRID_SHORTCUTS_ID}")`
to
`function=action_select_ag_grid_rows()`

```suggestion
@capture("action")
def action_select_ag_grid_rows(_trigger):
    # Sleep to see what part of the output component is updating
    time.sleep(0.5)

    num_rows = _trigger % 4
    selected_rows = df_3_rows.iloc[:num_rows].to_dict("records") if num_rows > 0 else []

    return selected_rows
```